File: prototypes/01-llm-connector/backend/src/mock/llm/chatgpt.mock.js
export function reply(type, data) {
  const name = (data.name || 'Inconnu').replace('-', ' ');
  const style = data.style || 'neutral';
  const length = data.length || 'medium';
  const llm = data.llm || 'chatgpt';
  const validType = ['biography', 'filmography', 'summary'].includes(type) ? type : 'contenu';

  return `Demande envoyée à ${llm} pour une ${validType} de "${name}", avec un style "${style}" et une longueur "${length}".`;
}

File: sandbox/rag-generator/backend-springboot/src/main/java/com/angular/starter/mock/llm/ChatGptMock.java
package com.angular.starter.mock.llm;

import org.springframework.stereotype.Component;

import java.util.Map;

@Component
public class ChatGptMock {

    public String reply(String type, Map<String, Object> input) {
        try {
            Thread.sleep(1000);
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }
        
        System.out.println("00000000001:reply:" + input);

        String name = normalize((String) input.getOrDefault("name", "Inconnu"));
        String style = (String) input.getOrDefault("style", "neutral");
        String length = (String) input.getOrDefault("length", "medium");
        String llm = "chatgpt";

        String validType;
        switch (type.toLowerCase()) {
            case "biography":
            case "filmography":
            case "summary":
                validType = type.toLowerCase();
                break;
            default:
                validType = "contenu";
        }

        return String.format(
            "Java Mock Backend SpringBoot - Demande envoyée à %s pour une %s de \"%s\", avec un style \"%s\" et une longueur \"%s\".",
            llm, validType, name, style, length
        );
    }

    private String normalize(String value) {
        return value.replace("-", " ");
    }
}
